src/odemis/model/_vattributes.py

```diff
@@ -982,6 +983,30 @@ def _check(self, value):
         Enumerated._check(self, value)
         VigilantAttribute._check(self, value)
 
+    def clip(self, val):
+        """ Clip the given value to fit within the choices.
+
+        If the range contains vectors of length n, each element of val will be
+        clipped separately by position for positions 0..n-1
+        return (same type as val): value clipped
+        """
+
+        if isinstance(val, tuple):
+            if val not in self.choices:
+                dist = distance.euclidean(val, max(self.choices))
+                clipped = max(self.choices)
+                for choice in self.choices:
+                    dist_curr = distance.euclidean(val, choice)
+                    # check if distance to choice smaller than previously found choice
+                    if all(dist < clipped):
+                        dist = dist_curr
+                        clipped = choice
+                return clipped
+            else:
+                return val
+        else:
+            raise ValueError("Value of %s not supported for VAEnumerated." % type(val))
```

Comment:
The idea of `clip` is that it always returns something acceptable (and never complains).
So instead of raising an error, I'd suggest that in the fallback code, it returns the current value.